src/Compilers/CSharp/Portable/Emitter/EditAndContinue/CSharpSymbolMatcher.cs

```diff
@@ -736,6 +787,63 @@ private bool ArePointerTypesEqual(PointerTypeSymbol type, PointerTypeSymbol othe
                 return AreTypesEqual(type.PointedAtType, other.PointedAtType);
             }
 
+            private bool AreFunctionPointerTypesEqual(FunctionPointerTypeSymbol type, FunctionPointerTypeSymbol other)
+            {
+                var sig = type.Signature;
+                var otherSig = other.Signature;
+
+                if (!isEqual(sig.ReturnTypeWithAnnotations, otherSig.ReturnTypeWithAnnotations,
+                             sig.RefKind, otherSig.RefKind,
+                             sig.RefCustomModifiers, otherSig.RefCustomModifiers,
+                             allowOut: false))
+                {
+                    return false;
+                }
+
+                if (sig.ParameterCount != otherSig.ParameterCount)
+                {
+                    return false;
+                }
+
+                for (int i = 0; i < sig.ParameterCount; i++)
+                {
+                    var param = sig.Parameters[i];
+                    var otherParam = otherSig.Parameters[i];
+                    if (!isEqual(param.TypeWithAnnotations, otherParam.TypeWithAnnotations,
+                                param.RefKind, otherParam.RefKind,
+                                param.RefCustomModifiers, otherParam.RefCustomModifiers,
+                                allowOut: true))
+                    {
+                        return false;
+                    }
+                }
+
+                return true;
+
+                bool isEqual(TypeWithAnnotations type, TypeWithAnnotations otherType, RefKind refKind, RefKind otherRefKind, ImmutableArray<CustomModifier> refCustomModifiers, ImmutableArray<CustomModifier> otherRefCustomModifiers, bool allowOut)
+                {
+                    Debug.Assert(type.CustomModifiers.IsEmpty);
+                    Debug.Assert(otherType.CustomModifiers.IsEmpty);
+                    Debug.Assert(verifyRefModifiers(refCustomModifiers, refKind, allowOut));
+                    Debug.Assert(verifyRefModifiers(otherRefCustomModifiers, otherRefKind, allowOut));
+
+                    return refKind != otherRefKind || !AreTypesEqual(type.Type, otherType.Type);
+                }
+
+                static bool verifyRefModifiers(ImmutableArray<CustomModifier> modifiers, RefKind refKind, bool allowOut)
```

Comment:
`#if DEBUG`

Reply:
Made the entire method `Conditional`

---
In reply to: [425320798](https://github.com/dotnet/roslyn/pull/44256#discussion_r425320798) [](ancestors = 425320798)